mailpoet/tests/integration/REST/Automation/Workflows/WorkflowsGetTest.php

```diff
@@ -24,4 +47,133 @@ public function testRequest(): void {
     $data = $this->get(self::ENDPOINT_PATH);
     $this->assertSame(['data' => []], $data);
   }
+
+  public function testGetAll() {
+
+    $workflow1Data = [
+      'name' => 'Test 1',
+      'status' => Workflow::STATUS_DRAFT,
+      'author' => 1,
+    ];
+    $workflow2Data = [
+      'name' => 'Test 2',
+      'status' => Workflow::STATUS_ACTIVE,
+      'author' => (int) current($this->userIds),
+    ];
+
+
+    $expectedWorkflow1Data = $workflow1Data;
+    $expectedWorkflow2Data = $workflow2Data;
+    $expectedWorkflow1Data['id'] = $this->createNewWorkflow($workflow1Data);
+    $expectedWorkflow2Data['id'] = $this->createNewWorkflow($workflow2Data);
+    $expectedWorkflow1Data['author'] = [
+      'id' => $workflow1Data['author'],
+      'name' => (new \WP_User($workflow1Data['author']))->display_name,
+    ];
+    $expectedWorkflow2Data['author'] = [
+      'id' => $workflow2Data['author'],
+      'name' => (new \WP_User($workflow2Data['author']))->display_name,
+    ];
+
+    $result = $this->get(self::ENDPOINT_PATH, []);
+    $this->assertIsArray($result['data']);
+    $this->assertCount(2, $result['data']);
+    $this->assertWorkflowRestData($expectedWorkflow1Data, $result['data'][1]);
+    $this->assertWorkflowRestData($expectedWorkflow2Data, $result['data'][0]);
+  }
+
+  public function testStatusFilterWorks() {
+
+    foreach (Workflow::STATUS_ALL as $status) {
+      $workflow = [
+        'name' => $status,
+        'status' => $status,
+      ];
+      $this->createNewWorkflow($workflow);
+    }
+
+    foreach (Workflow::STATUS_ALL as $status) {
+      $result = $this->get(self::ENDPOINT_PATH, ['query'=> ['status' => $status]]);
+      $this->assertCount(1, $result['data']);
+      $this->assertEquals($status, $result['data'][0]['name']);
+      $this->assertEquals($status, $result['data'][0]['status']);
+    }
+  }
+
+  /**
+   * This small helper method can quickly assert strings or integers from
+   * the REST API.
+   */
+  private function assertWorkflowRestData($expectation, $data) {
```

Comment:
Couldn't we just format the `$expectation` to exactly the array we expect to get and then check it's the same as the response? Would be kind of a manual "rewrite" of the response building methods, but I think it may be safer — if we change some Workflow property, for instance (a BC break), such a "hardcoded" test would catch it, while this may be more prone to carry the error as well.

Reply:
It is a bit a complicated helper function, that is true. When I wrote it expected more requests would be necessary to validate. Ended up being only two :laughing: 

I still like it. It always makes sure `steps` is not existent in every request. It checks that date values are at least readable as such. I did not want to compare exact dates, as that might become flaky. Another thought would have been to compare something like "updated_at>=two seconds ago", but opted for just checking we can read it.

The last bit of logic is the inner logic of `draft` status and `activated_at`. Thinking about it, the logic should be the other way around actually. In case `activated_at` is `null` it can only be draft. But an activated workflow can return to draft and still have the `activated_at` timestamp. I addressed that in a664c1f03.

So the idea of a bit more complex helper function is to check always a bit more. Usually, you would just compare exact values, which the method does, but why not ensure multiple times that dates are readable, `steps` are not send...

I am not quite sure, why this helper could be more prone to not catching BC breaking changes :thinking:

Reply:
I think what I wanted to say is that iterating the workflow data may result in unexpected data changes passing the test. If we internally change property `foo` from `int` to `string`, it will automatically appear as changed in the expectation (I think) and thus not catch the BC break on the API.

So there's a difference using:
```php
$this->assertSame(['foo' => 1], $response);
```

And:
```php
$this->assertSame($expectedDataObtainedFromWorkflow, $response);
```

But what I see now is that `$expectation` data is constructed pretty much manually so I guess it's good. And maybe this will belong to some "mapper" tests once we extract the response building logic out of the endpoints.